native/NativeSampleApps/SmartSyncExplorer/src/com/salesforce/samples/smartsyncexplorer/ui/MainActivity.java

```diff
@@ -238,39 +247,35 @@ private void launchAccountSwitcherActivity() {
 
 	@Override
 	public Loader<List<ContactObject>> onCreateLoader(int id, Bundle args) {
-		contactLoader = new ContactListLoader(this, curAccount);
+		contactLoader = new ContactListLoader(this, SmartSyncSDKManager.getInstance().getUserAccountManager().getCurrentUser());
 		return contactLoader;
 	}
 
 	@Override
 	public void onLoaderReset(Loader<List<ContactObject>> loader) {
-		originalData = null;
 		refreshList(null);
 	}
 
 	@Override
 	public void onLoadFinished(Loader<List<ContactObject>> loader,
 			List<ContactObject> data) {
-		originalData = data;
-		nameFilter.setOrigData(originalData);
 		refreshList(data);
 	}
 
 	@Override
 	public boolean onClose() {
-		refreshList(originalData);
 		return true;
 	}
 
 	@Override
 	public boolean onQueryTextSubmit(String query) {
-		filterList(query);
+		nameFilter.setFilterTerm(query);
```

Comment:
The NameFieldFilter remembers the filter term, setting the filter term cause the NameFieldFilter to update the ContactListAdapter.

Reply:
That way - when coming back to the list screen, whatever previously typed filter term is still applied.